### ISSUE TITLE
Formatter: deviation from Black when line splitting subscript assignment

### ISSUE THREAD

**Issue by rhynix** [astral-sh/ruff#9781]
Ruff splits lines differently when the line length exceeds the configured line length, and the line contains assignment using subscript notation. For instance, when assigning to a dict.

I had a look at the known deviations, and at other open issues, but could not find this deviation.

Input:

```python
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa["bbbbbbbbbbbbbbbbbbbb"] = "ccccccccccc"
```

Black output:

```python
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa["bbbbbbbbbbbbbbbbbbbb"] = (
    "ccccccccccc"
)
```

Ruff (`ruff format --isolated file.py`) output:

```python
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa[
    "bbbbbbbbbbbbbbbbbbbb"
] = "ccccccccccc"
```

Ruff version:

```
% ruff version
ruff 0.2.0
```

**Comment by rhynix** [comment]
Black used to use the style ruff currently uses, but was changed late 2022.

This was the issue: https://github.com/psf/black/issues/1498
This was the PR: https://github.com/psf/black/pull/3368

**Comment by MichaReiser** [comment]
Thanks for reporting this issue.

Is it possible that you use/used black 24? Or what black version are you using?

**Comment by rhynix** [comment]
> Is it possible that you use/used black 24? Or what black version are you using?

That is correct.

```
% black --version
black, 24.1.1 (compiled: no)
Python (CPython) 3.8.18
```

**Comment by rhynix** [comment]
I can confirm this cange was introduced in Black 24.1:

https://github.com/psf/black/blob/main/CHANGES.md#2410

> If an assignment statement is too long, we now prefer splitting on the right-hand side (#3368)

**Comment by dhruvmanila** [comment]
So, Black 24.1 moved a lot of preview style rules into it's stable style (https://github.com/psf/black/issues/4042). We've implemented the same in our own preview style formatting which can be enabled through `--preview` flag. Refer to the [preview docs](https://docs.astral.sh/ruff/preview/) for more details.

If you run the mentioned code snippet under the preview style then you'd get the same output:
```console
ruff format --preview --diff
```

```diff
--- /Users/dhruv/playground/ruff/formatter/isolated.py
+++ /Users/dhruv/playground/ruff/formatter/isolated.py
@@ -1 +1,3 @@
-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa["bbbbbbbbbbbbbbbbbbbb"] = "ccccccccccc"
+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa["bbbbbbbbbbbbbbbbbbbb"] = (
+    "ccccccccccc"
+)
```

You can track the progress of the pull request which will promote the same preview style rules into stable style for Ruff here: https://github.com/astral-sh/ruff/pull/9639

I hope this helps :)

**Comment by rhynix** [comment]
> So, Black 24.1 moved a lot of preview style rules into it's stable style ([psf/black#4042](https://github.com/psf/black/issues/4042)). We've implemented the same in our own preview style formatting which can be enabled through `--preview` flag. Refer to the [preview docs](https://docs.astral.sh/ruff/preview/) for more details.
> 
> If you run the mentioned code snippet under the preview style then you'd get the same output:
> 
> ```
> ruff format --preview --diff
> ```
> 
>  ```diff
> --- /Users/dhruv/playground/ruff/formatter/isolated.py
> +++ /Users/dhruv/playground/ruff/formatter/isolated.py
> @@ -1 +1,3 @@
> -aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa["bbbbbbbbbbbbbbbbbbbb"] = "ccccccccccc"
> +aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa["bbbbbbbbbbbbbbbbbbbb"] = (
> +    "ccccccccccc"
> +)
> ```
> 
> You can track the progress of the pull request which will promote the same preview style rules into stable style for Ruff here: #9639
> 
> I hope this helps :)

It sure does. Thank you for your help.

I'll close this issue, since I believe it does not provide any extra value at this point.
